docs/decision-records/wp-cli.md

```diff
@@ -0,0 +1,185 @@
+4. Run command ```wp --require=axioned.php setup```
+
+## Use Cases of WP-CLI
+
+- Manage plugins and themes
+    - Installing the WordPress software, themes, and/or plugins
+    - Activating/deactivating/updating themes and plugins
+    - Resolving conflicts between plugins and themes
+    - Creating a child theme
+    - Updating core WordPress via WP CLI
+- Quick security check for wordpress core and plugins
+- Usage in deployment scripts
+    - Search and Replace Command
+- Create loads of dummy content for testing
+- User management via WP CLI
+    - Creating or managing users, including changing passwords and other user information
+- Configuring WP-Cron to run like a real cron via WP CLI for stability
+    - Test WP Cron spawning system is running properly
+    - Run, schedule, delete, reschedule posts
+- Editing or optimizing your database, such as deleting post revisions
+- Generating test data 
+- Add your own command
+- Managing comments
+- Custom command in WP-CLI
+- DB Migration
+- Run commands on the remote server
+
+## Some useful commands:
+
+**1. WP Installation**
+
+```
+// download core wp files
+wp core download 
+
+// create config file
+wp core config --dbname=wpcli_test --dbuser=phpmyadmin --dbpass=root --dbhost=localhost --dbprefix=wp_
+
+// create db
+wp db create
+
+// install wp with admin user
+wp core install --url=http://localhost/WP-Cli_Testing/wp --title="Your Blog Title" --admin_name=admin --admin_password=admin --admin_email=admin@example.com
+
+//upgrade or downgrade wp core 
+wp core update --version=5.8.4 --force
+```
+
+Reinstall WordPress Core
+
+```
+wp core download --skip-content --force
+```
+
+Change home and site url
+
+```
+wp option update home 'https://example.com'
+wp option update siteurl 'https://example.com'
+```
+
+**2. Plugins play around**
+
+```
+wp plugin list
+
+wp plugin install all-in-one-wp-migration
+wp plugin delete all-in-one-wp-migration
+
+wp plugin activate hello
+wp plugin deactivate hello
+
+// install & activate
+wp plugin install all-in-one-wp-migration --activate
+
+// install specific version
+wp plugin install all-in-one-wp-migration --version=7.60
+```
+
+**3. Db Migration**
+import, export and search replace
+
+```
+wp db import [<file name>]
+wp db export
+wp search-replace http://example.com http://newexample.com
+```
+
+**4. User Management**
+```
+wp user list 
+wp user get admin
+
+// create user 
+wp user create ndg ndg@xyz.com --role=editor --first_name=alfa --last_name=beta --user_pass=admin
+
+// manage user capabilities 
+wp user list-caps ndg
+
+wp user add-cap ndg edit_plugins
+wp user remove-cap ndg edit_plugins
+
+// manage user role
+wp user add-role ndg administrator
+wp user remove-role ndg administrator
+
+wp user delete ndg
+```
+
+**5. Security check**
+
+Core files security check:
+
+```
+wp core verify-checksums --version=$(wp core version)
+```
+
+Plugins security check:
+```
+wp plugin verify-checksums --all
+```
+
```

Comment:
The custom command topic is not covered which we used for installing the setup in our WordPress starter repo

can you please add about that how we can create and use custom CLI commands